cmd/gokr-packer/packer.go

```diff
@@ -203,8 +204,20 @@ func findBuildFlagsFiles() (map[string]string, error) {
 			return nil, err
 		}
 
+		var buildFlags []string
+		sc := bufio.NewScanner(strings.NewReader(string(b)))
```

Comment:
Why not just `buildFlags := strings.Split(string(b), "\n")`?

Removing trailing spaces might not actually be desired in all cases. Think of an IRC bot which might get a `"--message_response_prefix=  (bot)  "` argument.

Reply:
> Why not just buildFlags := strings.Split(string(b), "\n")?

Breaks for people editing their stuff on Windows

> Removing trailing spaces might not actually be desired in all cases

I see. If we don't trim then things like I did in the Makefile might easily break:

	PKG_EVCC_BUILD_FLAGS := -tags=release -ldflags \
		-X github.com/andig/evcc/server.Version=$(VERSION) -X github.com/andig/evcc/server.Commit=$(SHA)

if I'm not mistaken then even the trailing ` ` after ldflags is wrong as it would have been stripped by the shell?

Is not trimming really correct or also error-prone? Alternatively, we could go back to shlex.

Reply:
> Breaks for people editing their stuff on Windows
> 
> >

Okay, that’s fair. Let’s stick to bufio.NewScanner then, but handle the error (the docs don’t indicate that we can skip the error check when using a strings.NewReader, or did I miss something?), and remove the trimming.

Your Makefile should generate an env file that doesn’t need trimming. You can use echo for that, I think, to get shell expansion if you need it.

Reply:
> but handle the error (the docs don’t indicate that we can skip the error check when using a strings.NewReader

You've lost me. `NewReader` doesn't error? Else I couldn't have put it inline?

Reply:
> You've lost me. `NewReader` doesn't error? Else I couldn't have put it inline either?

No, but you’re supposed to call and handle http://golang.org/pkg/bufio/#Scanner.Err after Scan() returns false :)

Reply:
Ahh, thanks for coaching me through. First time I've actually used bufio at all. Updated accordingly.